.circleci/config.yml

```diff
@@ -0,0 +1,154 @@
+version: 2.1
+
+executors:
+  default:
+    working_directory: /go/src/github.com/mattermost/
+    docker:
+      - image: mattermost/mattermost-build-webapp:oct-2-2018
+  ubuntu:
+    working_directory: ~/go/src/github.com/mattermost/mattermost-server
+    machine:
+      image: "ubuntu-1604:201903-01"
+    environment:
+      COMPOSE_PROJECT_NAME: "circleci"
+
+jobs:
+  setup:
+    executor:
+      name: default
+    steps:
+      - run: |
+          ls -la
+          mkdir -p ~/.ssh/
+          echo -e "Host github.com\n\tStrictHostKeyChecking no\n" > ~/.ssh/config
+          git clone git@github.com:mattermost/mattermost-webapp.git
+          cd mattermost-webapp
+          git checkout $CIRCLE_BRANCH || git checkout master
+          export WEBAPP_GIT_COMMIT=$(git rev-parse HEAD)
+          echo "$WEBAPP_GIT_COMMIT"
+          curl -f -o ./dist.tar.gz https://releases.mattermost.com/mattermost-webapp/commit/${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz && mkdir ./dist && tar -xvf ./dist.tar.gz -C ./dist --strip-components=1 || make node_modules test build
+      - persist_to_workspace:
+          root: /go/src/github.com/mattermost
+          paths:
+            - mattermost-server
+            - mattermost-webapp
+
+  check-i18n:
+    executor:
+      name: ubuntu
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/go/src/github.com/mattermost/
+      - run:
+          name: Check i18n
+          command: |
+            docker run -it \
+              -v ~/go/src:/go/src \
+              -w /go/src/github.com/mattermost/mattermost-server \
+              mattermost/mattermost-build-server:feb-28-2019 \
+              bash -c 'cd /go/src/github.com/mattermost/mattermost-server; cp i18n/en.json /tmp/en.json; make i18n-extract; diff /tmp/en.json i18n/en.json;'
+  build:
```

Comment:
For the build step maybe makes more sense to use the docker instead of the machine as executor
you can do this
```
   build:
    docker:
      - image: mattermost/mattermost-build-server:feb-28-2019
    working_directory: /go/src/github.com/mattermost/mattermost-server
    steps:
      - attach_workspace:
          at: /go/src/github.com/mattermost/
      - run:
          command: |
            make config-reset
            make check-style BUILD_NUMBER='${CIRCLE_BRANCH}-${CIRCLE_BUILD_NUM}'
            GOFLAGS=-p=8 make build BUILD_NUMBER='${CIRCLE_BRANCH}-${CIRCLE_BUILD_NUM}'
            GOFLAGS=-p=8 make package BUILD_NUMBER='${CIRCLE_BRANCH}-${CIRCLE_BUILD_NUM}'
      - store_artifacts:
          path: /go/src/github.com/mattermost/mattermost-server/dist/mattermost-team-linux-amd64.tar.gz
      - store_artifacts:
          path: /go/src/github.com/mattermost/mattermost-server/dist/mattermost-team-osx-amd64.tar.gz
      - store_artifacts:
          path: /go/src/github.com/mattermost/mattermost-server/dist/mattermost-team-windows-amd64.zip
      - persist_to_workspace:
          root: /go/src/github.com/mattermost
          paths:
            - mattermost-server
            - mattermost-webapp
```